src/loading2/octree-geometry-node.ts

```diff
@@ -15,8 +15,9 @@ export class OctreeGeometryNode implements IPointCloudTreeNode {
 
 	loaded: boolean = false;
 	loading: boolean = false;
+	failed: boolean = false;
 	parent: OctreeGeometryNode | null = null;
-	geometry: BufferGeometry | null = null;
+	geometry: BufferGeometry | undefined;
```

Comment:
Could you explain a bit why we're changing the type here?

Reply:
The main reason is the `toTreeNode` method of the `PointCloudOctree` class. 

I changed the `toTreeNode` method to use the updated `IPointCloudGeometryNode` interface. Then in that method, when calling the `Points()` constructor we can pass in undefined geometry which allows ThreeJS to still properly initialize that class (which is not possible with null)

Reply:
oh, that's nice! Thanks for the explanation.